isobus/src/can_message.cpp

```diff
@@ -326,4 +326,48 @@ namespace isobus
 		return (get_uint8_at(byteIndex) & mask) == mask;
 	}
 
+	std::uint64_t CANMessage::get_data_custom_length(const std::uint32_t startBitIndex, const std::uint32_t length, const isobus::CANMessage::ByteFormat format) const
+	{
+		std::uint64_t retVal = 0;
+		std::uint8_t currentByte = 0;
+		std::uint32_t endBitIndex = startBitIndex + length - 1;
+		std::uint32_t bitCounter = 0;
+		std::uint32_t amountOfBytesLeft = (length + 8 - 1) / 8;
+		std::uint32_t startAmountOfBytes = amountOfBytesLeft;
+		std::uint8_t indexOfFinalByteBit = 7;
+
+		for (auto i = startBitIndex; i <= endBitIndex; i++)
+		{
+			auto byteIndex = i / 8;
+			auto bitIndexWithinByte = i % 8;
+			auto bit = (data[byteIndex] >> (indexOfFinalByteBit - bitIndexWithinByte)) & 1;
```

Comment:
@ad3154 Yes, I will change to "at". I am not a fan to include try catch in the library function code, do you think it would be better to comment that you have to use try catch for a potential out of boundary exception when using this function?

Reply:
Or, as we know beforehand the full length of data and we know the length that is requested, we could easily perform boundary checking at the very beginning. It would be faster than having an overhead "at" in this case